### DISCUSSION TITLE
Does a call to an API function also wait for contract termination to resolve the promise ?

### DISCUSSION THREAD

**Discussion by AtegyekaRodgers** [reach-sh/reach-lang#990] [Q&A] [answered]
I would like to keep calling a number of API function while still attached to the same contract. 
How can I achieve this. 

In case you want to look at my code, check lines 28 - 36 inside src/dashboard.js
https://github.com/Reach-Insurance/react/blob/main/src/dashbord.js
![apis-halt](https://user-images.githubusercontent.com/47475700/168597422-611e391b-e739-443f-a952-4eb52a436d16.png)

**Comment by jeapostrophe** [comment, marked as answer]
Your code looks totally fine. It will make one call and then return. If you use `insurerContractHandle` twice, it will only do contract verification once. What do you think the problem is?

**Reply by AtegyekaRodgers** [reply]
Ok, Let me try more troubleshooting techniques first. I will let you know if I discover anything else
